### PR TITLE
Final

### PR DESCRIPTION
**Pull request by AckonSamuel**
Final Budget App
Note: Please check the README for instructions on installing and running the project
[Here](https://user-images.githubusercontent.com/92922987/202872265-9a56cd73-1ecd-4d03-9261-e626928d8426.mp4) is a short video demo for the app

For this Pr, I implement the following.

Splash screen
    - A simple page with the app's name, and links to the signup and log-in pages.

Sign-up and login pages
    -The user can register in the app with full name, email, and password (all mandatory).
    -The user can log into the app using email and password.
    -If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

Home page 
   - When the user logs in, they are presented with the group's page.
    -For each group, the user can see their name, icon, and the total amount of all the transactions that belong to that group.
    -When the user clicks (or taps) on a group item, the application navigates to the transactions page for that group.
    -There is a button "add new groups" at the bottom that brings the user to the page to create new group.

Expenditure page
    -For a given category, the list of Expenditures is presented, ordered by the most recent.
    -At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of Expenditures in that category).
    -There is a button "add a new Expenditure" at the bottom that brings the user to the page to create a new Expenditure.
    -When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new group" page
    -The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    -The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    -When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new Expenditure" page
    -The user fills out a form to create a new transaction with:
        name (mandatory)
        amount (mandatory)
        categories (required at least one)
    -The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the Expenditures page for that category.
    -When the user clicks on the "Back" button (<), the user navigates to the Expenditure page for that category.

# Note to Reviewer ❗❗❗❗❗
<hr>
Please, if you run Windows OS, add `gem 'ffi', github: 'ffi/ffi', submodules: true` to the gem file before you run bundle install, if you run into errors during the test rspec

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,101 @@
...
 
-* Ruby version
+- Ruby on Rails
+- postgresql
 
-* System dependencies
+## Live demo
 
-* Configuration
+[Live link](Coming soon...)
```
Comment (t-yanick) [inline README.md]: - [x] As is a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md)for  this project, " `Technical requirements: The project should be deployed and accessible online.` " you should deploy your application using Heroku, Netlify or GitHub Pages, make sure your project is deployed properly, the home and detail page works as expected, a  link to the deployed project should be present in the README.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @AckonSamuel, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark:  Good job with test coverage
:heavy_check_mark: Clean and organized code 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (AckonSamuel) [reply README.md]: please check again, @t-yanick 
I added the live link

**Review by t-yanick — APPROVED** [review]
Hi @AckonSamuel, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AckonSamuel** [comment]
Thank you @t-yanick

**Pull request merged**